src/coreclr/jit/emitarm64.h

```diff
@@ -316,14 +316,59 @@ static code_t insEncodeReg_Vm(regNumber reg);
 // Returns an encoding for the specified register used in the 'Va' position
 static code_t insEncodeReg_Va(regNumber reg);
 
-// Returns an encoding for the specified register used in the 'Pd' position
-static code_t insEncodeReg_Pd(regNumber reg);
+// Return an encoding for the specified register used in '4' thru '0' position.
+static code_t insEncodeReg_V_4_to_0(regNumber reg);
 
-// Returns an encoding for the specified register used in the 'Pn' position
-static code_t insEncodeReg_Pn(regNumber reg);
+// Return an encoding for the specified register used in '9' thru '5' position.
+static code_t insEncodeReg_V_9_to_5(regNumber reg);
 
-// Returns an encoding for the specified register used in the 'Pm' position
-static code_t insEncodeReg_Pm(regNumber reg);
+// Return an encoding for the specified register used in '12' thru '10' position.
+static code_t insEncodeReg_P_12_to_10(regNumber reg);
+
+// Return an encoding for the specified register used in '21' thru '17' position.
+static code_t insEncodeReg_V_21_to_17(regNumber reg);
+
+// Return an encoding for the specified register used in '21' thru '17' position.
+static code_t insEncodeReg_R_21_to_17(regNumber reg);
+
+// Return an encoding for the specified register used in '9' thru '5' position.
+static code_t insEncodeReg_R_9_to_5(regNumber reg);
+
+// Return an encoding for the specified register used in '4' thru '0' position.
+static code_t insEncodeReg_R_4_to_0(regNumber reg);
+
+// Return an encoding for the specified register used in '20' thru '17' position.
+static code_t insEncodeReg_P_20_to_17(regNumber reg);
+
+// Return an encoding for the specified register used in '3' thru '0' position.
+static code_t insEncodeReg_P_3_to_0(regNumber reg);
+
+// Return an encoding for the specified register used in '8' thru '5' position.
+static code_t insEncodeReg_P_8_to_5(regNumber reg);
+
+// Return an encoding for the specified register used in '13' thru '10' position.
+static code_t insEncodeReg_P_13_to_10(regNumber reg);
+
+// Return an encoding for the specified register used in '18' thru '17' position.
+static code_t insEncodeReg_R_18_to_17(regNumber reg);
+
+// Return an encoding for the specified register used in '7' thru '5' position.
+static code_t insEncodeReg_P_7_to_5(regNumber reg);
+
+// Return an encoding for the specified register used in '3' thru '1' position.
+static code_t insEncodeReg_P_3_to_1(regNumber reg);
+
+// Return an encoding for the specified register used in '2' thru '0' position.
+static code_t insEncodeReg_P_2_to_0(regNumber reg);
+
+// Return an encoding for the specified register used in '19' thru '17' position.
+static code_t insEncodeReg_V_19_to_17(regNumber reg);
```

Comment:
Should all the `V` ones be `Z`?

Reply:
I haven't introduced any "Z" terminology except for displaying the disassembly so kept it as `insEncodeReg_V*`. Even we use `REG_V0`, etc. Perhaps I should add aliases for "Z" registers and use it as well as rename methods whereever necessary.

Reply:
My preference would be Z, and introduce `REG_Z0` etc to make the code easier to read.
Same with `REG_P0`.

Reply:
Done for https://github.com/dotnet/runtime/pull/95129